saga-core/src/main/java/io/servicecomb/saga/core/application/SagaExecutionComponent.java

```diff
@@ -122,17 +125,57 @@ public void reanimate() {
     }
   }
 
-  private CompositeSagaLog compositeSagaLog(SagaLog sagaLog) {
+  private CompositeSagaLog compositeSagaLog(SagaLog sagaLog, PersistentStore persistentStore) {
     return new CompositeSagaLog(sagaLog, persistentStore);
   }
 
   private Map<String, SagaTask> sagaTasks(String sagaId, String requestJson, EventStore sagaLog) {
-    SagaLog compositeSagaLog = compositeSagaLog(sagaLog);
+    SagaLog compositeSagaLog = compositeSagaLog(sagaLog, persistentStore);
 
     return new HashMap<String, SagaTask>() {{
       put(SAGA_START_TASK, new SagaStartTask(sagaId, requestJson, compositeSagaLog));
-      put(SAGA_REQUEST_TASK, new RequestProcessTask(sagaId, compositeSagaLog, fallbackPolicy));
-      put(SAGA_END_TASK, new SagaEndTask(sagaId, compositeSagaLog));
+      SagaLog retrySagaLog = compositeSagaLog(sagaLog, SagaExecutionComponent.this.retrySagaLog);
+      put(SAGA_REQUEST_TASK, new RequestProcessTask(sagaId, retrySagaLog, fallbackPolicy));
+      put(SAGA_END_TASK, new SagaEndTask(sagaId, retrySagaLog));
     }};
   }
+
+  static class RetrySagaLog implements PersistentStore {
+
+    private final PersistentStore retryPersistentStore;
+    private final int retryDelay;
+
+    RetrySagaLog(PersistentStore persistentStore, int retryDelay) {
+      this.retryPersistentStore = persistentStore;
+      this.retryDelay = retryDelay;
+    }
+
+    @Override
+    public void offer(SagaEvent sagaEvent) {
+      boolean success = false;
+      do {
+        try {
+          retryPersistentStore.offer(sagaEvent);
+          success = true;
+        } catch (Exception e) {
+          e.printStackTrace();
+          try {
+            sleep(retryDelay);
+          } catch (InterruptedException e1) {
+            e1.printStackTrace();
```

Comment:
when this exception is thrown, normally it means this application is shutting down, because we didn't manually interrupt threads anywhere in code.

when saga is shutting down, we have to exit the while loop, or saga won't be able to stop. you may refer to FallbackPolicy on how to handle this exception.